_includes/_code/lakes-interactive/main.dart

```diff
@@ -0,0 +1,157 @@
+import 'package:flutter/material.dart';
+// Uncomment lines 3 and 6 to view the visual layout at runtime.
+import 'package:flutter/rendering.dart';
+
+void main() {
+  //debugPaintSizeEnabled = true;
+  runApp(new MyApp());
+}
+
+class FavoriteWidget extends StatefulWidget {
+  @override
+  _FavoriteWidgetState createState() => new _FavoriteWidgetState();
+}
+
+class _FavoriteWidgetState extends State<FavoriteWidget> {
+  bool _isFavorited = true;
+  int _favoriteCount = 41;
+
+  void toggleFavorite() {
+    setState(() {
+      // If the lake is currently favorited, unfavorite it.
+      if (_isFavorited) {
+        _favoriteCount--;
```

Comment:
I would prefer `-= 1` and `+= 1` but that's purely a personal thing, feel free to ignore.